mitosheet/src/mito/Mito.tsx

```diff
@@ -516,6 +540,33 @@ export const Mito = (props: MitoProps): JSX.Element => {
                     dfName={sheetDataArray[uiState.currOpenModal.sheetIndex] ? sheetDataArray[uiState.currOpenModal.sheetIndex].dfName : 'this dataframe'}
                 />
             )
+            case ModalEnum.UserEditedCode: return (
+                <UserEditedCodeModal
+                    setUIState={setUIState}
+                    mitoAPI={mitoAPI}
+                    userProfile={userProfile}
+                    onClickButton={(overwriteUserEdits: boolean) => {
+                        props.jupyterUtils?.writeGeneratedCodeToCell(
+                            analysisData.analysisName, 
+                            analysisData.code, 
+                            userProfile.telemetryEnabled, 
+                            analysisData.publicInterfaceVersion, 
+                            () => {
+                                setUIState(prevUIState => {
+                                    return {
+                                        ...prevUIState,
+                                        currOpenModal: {
+                                            type: ModalEnum.UserEditedCode,
+                                        }
+                                    }
+                                })
+                            },
+                            oldCodeRef?.current,
+                            overwriteUserEdits,
+                        )
+                    }}
```

Comment:
This feels over abstracted to me. Why not just put this logic in the component itself? It's not like that component is ever going to do something different. Let's move it in by just making the `jupyterUtils` the prop to this component, and making this call in the component.